Sources/Web3Core/Structure/Block/Block.swift

```diff
@@ -102,6 +102,40 @@ extension Block: Decodable {
             return data
         }
     }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encodeHex(number, forKey: .number)
+        try container.encodeHex(hash, forKey: .hash)
+        try container.encodeHex(parentHash, forKey: .parentHash)
+        try container.encodeHexIfPresent(nonce, forKey: .nonce)
+        try container.encodeHex(sha3Uncles, forKey: .sha3Uncles)
+        
+        try container.encodeHexIfPresent(logsBloom?.bytes, forKey: .logsBloom)
+
+        try container.encodeHex(transactionsRoot, forKey: .transactionsRoot)
+        try container.encodeHex(stateRoot, forKey: .stateRoot)
+        try container.encodeHex(receiptsRoot, forKey: .receiptsRoot)
+
+        try container.encodeIfPresent(miner?.address, forKey: .miner)
+
+        try container.encodeHex(difficulty, forKey: .difficulty)
+        try container.encodeHex(totalDifficulty, forKey: .totalDifficulty)
+        try container.encodeHex(extraData, forKey: .extraData)
+        try container.encodeHex(size, forKey: .size)
+        try container.encodeHex(gasLimit, forKey: .gasLimit)
+        try container.encodeHex(gasUsed, forKey: .gasUsed)
+
+        // optional, since pre EIP-1559 block haven't such property.
+        try container.encodeHexIfPresent(baseFeePerGas, forKey: .baseFeePerGas)
+
+        try container.encodeHex(timestamp, forKey: .timestamp)
+
+        try container.encode(transactions, forKey: .transactions)
+
+        try container.encode(uncles.map { $0.hexString }, forKey: .uncles)
+
+    }
```

Comment:
Please, write test cases for that feature.

Reply:
I suggest the following:
- a test case where you create a Block manually, encode, decode it back and compare it with the original;
- a test case where you will have a JSON string, decode it and check that the decoding didn't throw. That should be something like `XCTAssertNoThrow(try JSONDecoder().decode(jsonString, Block.self))`.

To produce a JSON string use the first test and `String.init(data: try! JSONEncoder().encode(block), encoding: .utf8)!` and print it out to the console.